src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7743,11 +7762,16 @@ private ForStatementSyntax ParseForStatement()
                 }
                 else
                 {
-                    isDeclaration = !this.IsQueryExpression(mayBeVariableDeclaration: true, mayBeMemberDeclaration: false) &&
-                                    this.ScanType() != ScanTypeFlags.NotType &&
-                                    this.IsTrueIdentifier();
+                    decl = TryParseDeconstructionDeclaration(withEquals: true);
 
-                    this.Reset(ref resetPoint);
+                    if (decl == null)
+                    {
+                        isDeclaration = !this.IsQueryExpression(mayBeVariableDeclaration: true, mayBeMemberDeclaration: false) &&
+                                        this.ScanType() != ScanTypeFlags.NotType &&
+                                        this.IsTrueIdentifier();
+
+                        this.Reset(ref resetPoint);
+                    }
```

Comment:
Please add a test for this scenario.

---
In reply to: [68144036](https://github.com/dotnet/roslyn/pull/12104#discussion_r68144036) [](ancestors = 68144036)